python/ql/src/experimental/typetracking/TypeTracker.qll

```diff
@@ -88,6 +89,28 @@ module StepSummary {
     )
   }
 
+  pragma[noinline]
+  private predicate smallstepCall(Node nodeFrom, LocalSourceNode nodeTo, StepSummary summary) {
+    callStep(nodeFrom, nodeTo) and summary = CallStep()
+    or
+    returnStep(nodeFrom, nodeTo) and
+    summary = ReturnStep()
+  }
+
+  /**
+   * Gets the summary that corresponds to having taken a forwards
+   * local, heap and/or inter-procedural step from `nodeFrom` to `nodeTo`.
+   *
+   * Unlike `StepSummary::step`, this predicate does not compress
+   * type-preserving steps.
+   */
+  pragma[inline]
+  predicate smallstep(Node nodeFrom, LocalSourceNode nodeTo, StepSummary summary) {
+    smallstepNoCall(nodeFrom, nodeTo, summary)
```

Comment:
I must admit, the pidgin-like quality of `smallstepNoCall` bugs me a bit. Perhaps `smallstepWithoutCall` would be better (ah, but then one might think this should only exclude call steps, and not return steps -- though this criticism also applies to `smallstepNoCall`). 

Alternatively, `smallstepCall` could be come `interproceduralStep`, but then `smallstepNoCall` is not quite `intraproceduralStep` because it includes jump-steps. `nonInterproceduralStep`? Bleh. :neutral_face: 

Maybe `smallstepNoCall` isn't too bad after all...

Reply:
Let me know if you come up with a better name ;-)